### DISCUSSION TITLE
Ollama API endpoint Connection Issue

### DISCUSSION THREAD

**Discussion by Natedorr** [open-webui/open-webui#6388] [Issues]
Using the ollama api endpoint has had issues when using with continue.dev since the 0.3.22 release.  My config.json for continue.dev looks like this:

    {
      "model": "AUTODETECT",
      "title": "Ollama-Bad",
      "provider": "ollama",
      "apiBase": "http://localhost:8080/ollama/",
      "contextLength": 7999,
      "completionOptions": {},
      "requestOptions": {
        "headers": {
          "Authorization": "Bearer sk-xxx"
        }
      }
    },

This works with open-webui 0.3.21 as well as using the localhost:11434 port.  When i try to chat in the vscode continue.dev plugin, I get the models populating properly, but there is no response.  It spins and then returns empty.    I pulled the following from the vscode output for 'Continue - LLM Prompt/Completion'

```
==========================================================================
==========================================================================
Settings:
contextLength: 7999
model: nemotron:70b-instruct-q3_K_M
maxTokens: 1024
stop: <|start_header_id|>,<|end_header_id|>,<|eot_id|>
log: undefined

############################################

<user>
Are you working

==========================================================================
==========================================================================
Completion:



```

This error has been happening with multiple versions of the base ollama (separate install than open-webui).  Also i rolled back continue releases and still had the issue.
I have replicated this issue on Linux and a WSL windows environment.

**Comment by tjbck** [comment]
![image](https://github.com/user-attachments/assets/b62e49ad-88fb-45e1-b8f3-e51208f19087)

Unsure which endpoint continue.dev uses, but I can verify both the `/api/generate` and `/api/chat` works as intended here. Continue your troubleshooting journey, and keep us updated!

Related: https://docs.openwebui.com/api

**Reply by Natedorr** [reply]
good pun.  The main.py for ollama endpoints did get the /api/chat/ updated at line 818, but the only update was `stream=form_data.stream,` being added:

```
    return await post_streaming_url(
        f"{url}/api/chat",
        json.dumps(payload),
        stream=form_data.stream,
        content_type="application/x-ndjson",
    )
```

Continue is using the api/chat for the completions as well.  

Seeing if i can work through how to make continue modify the stream parameter to isolate further.

**Reply by Natedorr** [reply]
I think that previously the functionality was that since the post_streaming_url had stream default to true, now by passing the stream as form_data.stream with it defaulted to none, will make it default to false.  That line being updated to the following:

`stream=form_data.stream or True,`

would revert to the prior behavior of the method.  

As far as consistency is concerned, the ollama api\chat is defaulted to true.

**Reply by tjbck** [reply]
PR Welcome!

**Reply by Natedorr** [reply]
PR started here:

https://github.com/open-webui/open-webui/pull/6399
